docs/source/publishers.rst

```diff
@@ -0,0 +1,254 @@
+returning the result of the final publisher.
+
+All data returned by ``publish_alert()`` should be assumed as optional.
+
+.. code-block:: python
+
+  from stream_alert.alert_processor.helpers import publish_alert
+
+  def _dispatch(self, alert, descriptor):
+    # ...
+    publication = publish_alert(alert, self, descriptor)
+    make_api_call(misc_data=publication)
+
+
+**"Default" Implementations**
+
+For output-specific fields that are mandatory (such as an incident Title or assignee), each output
+should offer a default implementation:
+
+.. code-block:: python
+
+  def _dispatch(self, alert, descriptor):
+    default_title = 'Incident Title: #{}'.format(alert.alert_id)
+    default_html = '<html><body>Rule: {}</body></html>'.format(alert.rule_description)
+    # ...
+
+
+**Custom fields**
+
+To allow Publishers fine-grained control of the look and feel of alerts, outputs can be implemented to
+offer custom fields that can control.
+
+.. code-block:: python
+
+  def _dispatch(self, alert, descriptor):
+    # ...
+    publication = publish_alert(alert, self, descriptor)
+
+    title = publication.get('pagerduty.title', default_title)
+    body_html = publication.get('pagerduty.body_html', default_html)
+
+    make_api_call(title, body_html, data=publication)
+
+
+**Alert Fields**
+
+When outputs require mandatory fields that are not subject to publishers, they should reference the ``alert``
+fields directly:
+
+.. code-block:: python
+
+  def _dispatch(self, alert, descriptor):
+    rule_description = alert.rule_description
+    # ...
+
+
+Registering Publishers
+----------------------
+
+Register publishers on a rule using the ``publisher`` argument on the ``@Rule`` annotation:
+
+.. code-block:: python
+
+  from publishers import publisher_1, publisher_2
+
+  @Rule(
+    logs=['stuff'],
+    outputs=['pagerduty', 'slack'],
+    publishers=[publisher_1, publisher_2]
+  )
+  def my_rule(rec):
+    # ...
+
+The ``publishers`` argument is a structure containing references to **Publishers** and can follow any of the
+following structures:
+
+**String, Publisher Name**
+
+.. code-block:: python
+
+  publishers=publisher_1
+
+When using this syntax, the given publisher will be applied to all outputs.
+
+
+**List of Publisher Name Strings**
+
+.. code-block:: python
+
+  publishers=[publisher_1, publisher_2, publisher_3]
+
+When using this syntax, all given publishers will be applied to all outputs.
+
+
+**Dict mapping Output to Publisher Names**
+
+.. code-block:: python
+
+  publishers={
+    'pagerduty:analyst': [publisher_1, publisher_2],
+    'pagerduty': [publisher_3, publisher_4],
+    'demisto': other_publisher,
+  }
+
+When using this syntax, publishers under each key will be applied to their matching outputs. Publisher keys
+with specific outputs (e.g. ``pagerduty:analyst``) are loaded first, before publisher keys that pertain
+to groups of outputs (e.g. ``pagerduty``).
```

Comment:
![image](https://user-images.githubusercontent.com/13323701/53263005-0c876800-368d-11e9-9432-14f14ad6494b.png)

Reply:
maybe not? I guess I didn't see exactly how you implemented it, so it probably works either way?

Reply:
(Github chewed up my response message or something...)

I decided to flip it around because I agree with your argument as it does make more intuitive sense.